base/linalg/arnoldi.jl

```diff
@@ -237,7 +237,11 @@ function _eigs(A, B;
     end
 
     # Refer to ex-*.doc files in ARPACK/DOCUMENTS for calling sequence
-    matvecA(x) = A * x
+    if method_exists(Base.A_mul_B!, (Vector{T},typeof(A),Vector{T}))
```

Comment:
I change this but the problem now is that `A_mul_B!` won't work for custom types with `*` defined but without indexing (see the last failure).

The problem can be ultimately tracked down to `generic_matvecmul` which is defined in terms of indexing operations, but for a type that is an array container with only custom `*` overloaded it will fail.